lib/Service/SnoozeService.php

```diff
@@ -89,11 +164,26 @@ public function wakeMessages(): void {
 	 * @param int $unixTimestamp
 	 * @return void
 	 */
-	public function snoozeMessage(Message $message, int $unixTimestamp): void {
+	public function snoozeMessageDB(Message $message, int $unixTimestamp): void {
 		$snooze = new MessageSnooze();
 		$snooze->setMessageId($message->getMessageId());
 		$snooze->setSnoozedUntil($unixTimestamp);
-		$this->messageSnoozeMapper->insert($snooze);
+		try {
+			$this->messageSnoozeMapper->insert($snooze);
+		} catch(Exception $e) {
```

Comment:
can you check the reason of the error and only handle duplicates and rethrow the rest?

Reply:
Added a check `$e->getPrevious() instanceof \Doctrine\DBAL\Exception\UniqueConstraintViolationException`. Do you think this is okay?

Reply:
Apparently, tests fail because the UniqueConstraintViolationException is part of 3rdparty. Is there a way to use it without errors or is this not an option?

Reply:
``\OCP\DB\Exception::getReason`` and comparison with ``\OCP\DB\Exception::REASON_UNIQUE_CONSTRAINT_VIOLATION``